src/main/java/de/bonndan/nivio/model/Labeled.java

```diff
@@ -1,13 +1,149 @@
 package de.bonndan.nivio.model;
 
+import org.springframework.util.StringUtils;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
 /**
  * Anything that has labels (key-value).
- *
- *
  */
 public interface Labeled {
 
+    String PREFIX_VALUE_DELIMITER = ";";
+
+    default String getLabel(Label key) {
+        return getLabel(key.name().toLowerCase());
+    }
+
     String getLabel(String key);
 
+    /**
+     * Returns all label with values having the given prefix.
+     */
+    default Map<String, String> getLabels(String prefix) {
+        return Labeled.withPrefix(prefix, getLabels());
+    }
+
+    /**
+     * Returns all labels with the given prefix.
+     *
+     */
+    default Map<String, String> getLabels(Label prefix) {
+        return getLabels(prefix.name().toLowerCase());
+    }
+
+    default void setLabels(String prefix, String[] keys, String[] values) {
```

Comment:
Why are ``keys`` and ``values`` not passed as ``Map``? Because it looks as if that is expected.

Reply:
Method has been removed.